DriverDataTimelineUI/src/main/java/com/drivequant/drivekit/timeline/ui/component/roadcontext/RoadContextViewModel.kt

```diff
@@ -9,45 +9,44 @@ import com.drivequant.drivekit.common.ui.component.contextcard.DKContextCard
 import com.drivequant.drivekit.common.ui.component.contextcard.DKContextCardItem
 import com.drivequant.drivekit.common.ui.utils.DKDataFormatter
 import com.drivequant.drivekit.common.ui.utils.convertToString
-import com.drivequant.drivekit.databaseutils.entity.DKRawTimeline
+import com.drivequant.drivekit.databaseutils.entity.RoadContext
+import com.drivequant.drivekit.driverdata.timeline.DKDriverTimeline
 import com.drivequant.drivekit.timeline.ui.R
 import com.drivequant.drivekit.timeline.ui.component.roadcontext.enum.EmptyRoadContextType
 import com.drivequant.drivekit.timeline.ui.component.roadcontext.enum.TimelineRoadContext
-import com.drivequant.drivekit.timeline.ui.distanceByRoadContext
-import com.drivequant.drivekit.timeline.ui.hasData
-import com.drivequant.drivekit.timeline.ui.totalDistanceForAllContexts
+import com.drivequant.drivekit.timeline.ui.component.roadcontext.enum.toTimelineRoadContext
+import java.util.Date
 
 internal class RoadContextViewModel : ViewModel(), DKContextCard {
 
     val changeObserver: MutableLiveData<Any> = MutableLiveData()
 
     private var distanceByContext = mapOf<TimelineRoadContext, Double>()
-        set(value) {
-            field = value.filterNot { it.value <= 0.0 }
-        }
-
     private var contextCards: List<DKContextCardItem> = emptyList()
     private var totalDistanceForAllContext = 0.0
-    private var distance = 0.0
     private var emptyRoadContextType: EmptyRoadContextType? = null
 
-    fun configure(timeline: DKRawTimeline?, selectedIndex: Int?) {
-        if (timeline != null && selectedIndex != null && timeline.hasData()) {
-            this.totalDistanceForAllContext = timeline.totalDistanceForAllContexts(selectedIndex)
-            this.distanceByContext = timeline.distanceByRoadContext(selectedIndex)
-            this.distance = 0.0
-            this.distanceByContext.forEach {
-                this.distance += it.value
+    fun configure(timeline: DKDriverTimeline?, selectedDate: Date?) {
+        if (timeline != null && selectedDate != null) {
+            this.distanceByContext = buildMap {
+                timeline.roadContexts.forEach { roadContextItem ->
+                    val distance = roadContextItem.value.firstOrNull { it.date == selectedDate }?.distance ?: 0.0
+                    if (distance > 0) {
+                        this[roadContextItem.key.toTimelineRoadContext()] = distance
+                    }
+                }
             }
+
+            this.totalDistanceForAllContext = timeline.allContext.firstOrNull { it.date == selectedDate }?.distance ?: 0.0
+
             this.emptyRoadContextType = if (this.distanceByContext.isEmpty()) {
                 EmptyRoadContextType.NO_DATA
             } else {
                 null
             }
         } else {
             this.totalDistanceForAllContext = 0.0
-            this.distanceByContext = emptyMap()
-            this.distance = 0.0
+            this.distanceByContext = mutableMapOf()
```

Comment:
Désolé je n'avais pas vu cette ligne, on peut remplacer par `emptyMap()`
```suggestion
            this.distanceByContext = emptyMap()
```